independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/BootstrapAppModelFactory.java

```diff
@@ -212,6 +213,18 @@ public AppModelResolver getAppModelResolver() {
     }
 
     public CurationResult resolveAppModel() throws BootstrapException {
+        if (test) {
+            //gradle tests encode the result on the class path
+
+            try (InputStream existing = getClass().getResourceAsStream(BootstrapConstants.SERIALIZED_APP_MODEL)){
```

Comment:
Wouldn't it be safer and perhaps more efficient to look for that file in `appClasses`? Now that the file is created in the classes dir, it will end up in the jar, right? Just wondering where there could be multiple files on the classpath.

Reply:
It is created in the test class directory, so it should not end up in the jar. It does stay on the file system though so if you attempt to run a test from the IDE after a build it should still be picked up.